src/python/pants/backend/python/lint/ruff/skip_field.py

```diff
@@ -11,17 +11,31 @@
 from pants.engine.target import BoolField
 
 
-class SkipRuffField(BoolField):
-    alias = "skip_ruff"
+class SkipRuffCheckField(BoolField):
+    alias = "skip_ruff_check"
     default = False
-    help = "If true, don't run ruff on this target's code."
+    help = "If true, don't run the ruff checker on this target's code."
+
+    deprecated_alias = "skip_ruff"
```

Comment:
How does this feature work? When I substitute `skip_ruff` instead of `skip_ruff_check` in tests, the tests fail.

Reply:
Hm, good question. It seems like it sort-of works. When I run this PR via `PANTS_SOURCE`, on a repo with `skip_ruff` field on a file:

- the field is understood (i.e. no error about an unknown field, like for `skip_not_a_real_field`) ✅ 
- it doesn't do any skipping, i.e. ruff is still run ❌  (probably the same as your observation of tests failing).
- there's no deprecation warning ❌ 

Thus, maybe we instead have to have have two dedicated field classes, both the old and new, and check for them both explicitly. One might then be able to use `removal_version`/`removal_hint` for the deprecation 🤞 , or maybe one has to go all the way down to a `warn_or_error` call somewhere (not sure where).

Reply:
I suspect you found a bug where a target generator is not propagating a deprecated field properly. I'm investigating.. :)

Reply:
> Thus, maybe we instead have to have have two dedicated field classes, both the old and new, and check for them both explicitly.

Right. I can preserve `SkipRuffField` with its old behavior as @kaos suggested here: https://github.com/pantsbuild/pants/issues/20620#issuecomment-1966087173

Then, we don't have to deprecate anything 🤷

Reply:
Sounds good to me. Will still fix the underlying issue though. #20627

Reply:
Restored `skip_ruff` as an option and made it so that the check/format rules will skip if either `skip_ruff_check`/`skip_ruff_format` or `skip_ruff` are set.